### ISSUE TITLE
Unclear inferred version number

### ISSUE THREAD

**Issue by bergtwvd** [mooltiverse/nyx#129]
Before we use nyx in anger, we have set up a small test environment.
We use the extended preset and are confused in the inferred version number for both main and development branches. Could you explain how we get to the version numbers?

We also see an odd message in the (attached) debug log for the infer on the development branch:
``DEBU[0000] current repository status 'DIRTY' not matched by release type 'integration' matchWorkspaceStatus filter 'CLEAN'. Skipping release type 'integration'``

This hints to uncommitted files in the workspace, but there are none (i.e. clean status).
Could you explain?

Log file:  [infer-debug.txt](https://github.com/mooltiverse/nyx/files/10329504/infer-debug.txt)

Steps to reproduce:
````
From GitLab webUI:
1. create empty repo, with README file
2. create file .nyx.yml with JSON content: { "preset":"extended" }
3. create "development" branch

From workspace:
1. clone repo
2. switch to development branch

From other shell:
3. start container:  docker run --rm -it --entrypoint=sh -v $PWD/:/project mooltiverse/nyx
4. nyx --info
   INFO[0000] Version: '0.1.0'

From workspace, development branch:
5. commit and push file "test.txt" (content "hello") with commit message "feat: minor fix"

From container:
6. nyx --info
   INFO[0000] Version: '0.2.0-internal.1+timestamp.19700120083646'
   
From workspace:
7. switch to main branch
8. commit and push file "test2.txt" (content "hello2") with commit message "feat: minor fix2"

From container:
9. nyx --info
   INFO[0000] Version: '0.2.0-internal.1+timestamp.19700120083648'

From workspace:
10. switch to development branch

From container:
11. nyx --info
   INFO[0000] Version: '0.2.0-internal.1+timestamp.19700120083648'
````

**Comment by flelli** [comment]
Hi @bergtwvd ,

I guess you already hit the point as the issue is only about the workspace status, which is _DIRTY_. If you run `git status` in both branches you should see the pending (uncommitted) changes.
As an alternative you can override the release type configuration and remove the `matchWorkspaceStatus` filter do it doesn't prevent a certain release type to be matched if there are uncommitted changes.

**Comment by bergtwvd** [comment]
Hi, I am still puzzled how nyx gets the the name `internal` in the version.

I do a fresh clone of the repository and a git status:

![image](https://user-images.githubusercontent.com/10650187/210536416-3638646e-8779-48d9-9609-73b2c940c752.png)

In this example there is a main and development branch, but not tags have been set yet. The following files are committed:

![image](https://user-images.githubusercontent.com/10650187/210538898-48a5a669-411d-4e41-b1bc-08e3673ef57b.png)

Where `.nyx.yml` contains:

![image](https://user-images.githubusercontent.com/10650187/210539024-66a4dd39-306e-41bd-9c2e-c7afae69f7f5.png)

Next I start the container with nyx:

![image](https://user-images.githubusercontent.com/10650187/210536688-b1a9e6f3-faf5-4b82-a111-d907f78c9cdf.png)

And I run the command `nyx --info`:

![image](https://user-images.githubusercontent.com/10650187/210536829-5412ec9f-4373-435a-8f78-f6774c4607bf.png)

Looking in the log with debug I see that nyx marks the repo status as dirty, skipping mainline as branch, and it eventually ends up using internal as name:

![image](https://user-images.githubusercontent.com/10650187/210537527-8c539eba-94de-4085-9305-65f33be501c2.png)

I this how it is supposed to work?

**Comment by flelli** [comment]
Hi, that's the expected behavior unless Nyx fails to detect the repo status (CLEAN or DIRTY). Can you please run `git status` right before and after running Nyx?

**Comment by flelli** [comment]
After testing with different scenarios I've hit some cases where the CLEAN/DIRTY is wrong. I'm reopening this issue

**Comment by flelli** [comment]
After a quick investigation it turns out there's a bug. I opened #130  for this

**Comment by flelli** [comment]
Fix is generally available in release [1.3.3](https://github.com/mooltiverse/nyx/releases/tag/1.3.3).
